### DISCUSSION TITLE
Logging Requests and Responses

### DISCUSSION THREAD

**Discussion by charleskoko** [saloonphp/saloon#434] [Ideas]
It would be useful to add a feature that allows logging of the requests made (including the headers and the body of the requests) as well as the responses (the entire response body, headers, etc.).

There are scenarios in production where we need to know exactly what was sent and what response was received from the API to resolve an issue reported by a client. This would greatly facilitate debugging and incident analysis.

**Comment by sinnbeck** [comment]
You could write a plugin (trait) that just calls the debugger and then logs it.
https://docs.saloon.dev/the-basics/debugging#custom-debugger-handlers
